### DISCUSSION TITLE
fit point cloud of object rotate around y axis in superquadric

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/community#666] [Q&A] [answered]
Hi,

Recently i test icub-gazebo-grasping, and try to grasp a steer object, some times the test failed and get report "No good grasp candidates found!".

I don't know whether it's because the steer is rotate by y axis about 15 degree, that the superquadric didn't fit well, then the grasp algorithm didn't get good candidates.

As the [notes](https://github.com/robotology/find-superquadric#note) said, current model is supposed to be lying on a table parallel to the x-y plane (to keep things simple). Can i fit point cloud of object that rotate around y axis through some modification?

Thanks.

![ok](https://github.com/robotology/community/assets/7019692/34e1bb7e-ef85-4ded-b6fe-904a96274896)
![fail](https://github.com/robotology/community/assets/7019692/5a7e88b6-7868-4bbc-84bd-777ccbb95094)

![2023-11-04_06-41](https://github.com/robotology/community/assets/7019692/9da627da-299c-4d3c-8e92-2b09ae6adb21)

**Comment by pattacini** [comment, marked as answer]
Hi @jimingda 

>As the [notes](https://github.com/robotology/find-superquadric#note) said, current model is supposed to be lying on a table parallel to the x-y plane (to keep things simple). Can i fit point cloud of object that rotate around y axis through some modification?

You can apply a convenient mapping (i.e., a homogeneous transformation) earlier to the input data and the inverse mapping right after the fitting.

However, I don't think this is the point here. The SQ fitting as provided in that repo is clearly not a good option in this case. You don't have to recreate a model from the point cloud[^1] as you know already it. 

The problem you should address is rather the _3D pose estimation of known objects_.

You may want to check out [DOPE](https://docs.nvidia.com/isaac/archive/2020.2/packages/object_pose_estimation/doc/dope.html), for example.

[^1]: https://github.com/robotology/find-superquadric just deals with very simple geometries, whereas the steering wheel is definitely more complex.
